### PR TITLE
[_6465] allow NREP remote execution hints to target INST_NAME (4-2-stable)

### PR DESCRIPTION
**Pull request by d-w-moore**
Includes tests, which are written for the Python RE plugin suite
and append the NREP to test behavior with multiple rule engine plugins
loaded.

### CODE REVIEW

File: scripts/irods/test/test_rule_engine_plugin_framework.py [hunk 1]
```diff
@@ -81,6 +85,86 @@ def test_bulk_put_operation__5164(self):
...
+                    self.assertTrue(
+                        lib.log_message_occurrences_equals_count(
+                            msg='Improperly formatted rule text in Python rule engine plugin',
+                            count=0,
+                            start_index=log_offset))
+
+
+    @unittest.skipIf(plugin_name != 'irods_rule_engine_plugin-python' or test.settings.RUN_IN_TOPOLOGY, "Skip for Python REP and Topology Testing")
+    def test_remote_python_rule_engine_plugin_instance_6465(self):
+        with append_native_re_to_server_config():
```
Comment (trel) [inline scripts/irods/test/test_rule_engine_plugin_framework.py:132]: is this correct?   append_native?

Comment (d-w-moore) [reply scripts/irods/test/test_rule_engine_plugin_framework.py:132]: Yes.. We are allowing native and PREP to co-exist in the configuration for purposes of the test.

**Comment by trel** [comment]
oh, looks like i didn't get that cen7 RPM signed correctly...

**Comment by trel** [comment]
yes, signed RPMs are happy now.

**Comment by korydraughn** [comment]
Do the tests pass?

**Comment by d-w-moore** [comment]
Yes, the only tests that failed were test_misc, something to do with server restart.  I believe it failed pre-commit too.

**Comment by alanking** [comment]
Probably related to: https://github.com/irods/irods/issues/6521

**Comment by d-w-moore** [comment]
Hm it was ... test_misc . test_server_respawns_processes__issue_4977

**Comment by alanking** [comment]
Oh. I still have reason to believe that failure is connected to the same root cause, but that is indeed a different test. It is also one that has been known to be flaky.

**Review by korydraughn — APPROVED** [review]
Let's get it merged!

**Comment by korydraughn** [comment]
If we don't have an issue to review `test_misc.test_server_respawns_processes__issue_4977`, we should go ahead and create that while it's fresh on our minds.

**Comment by d-w-moore** [comment]
Ok - done

**Comment by d-w-moore** [comment]
"Done" meant the #.  @korydraughn  would you like me to create that issue as well ?

**Comment by alanking** [comment]
@d-w-moore - Yes, please create the issue and link in a comment here when it's done. I was going to do it but I don't have the details handy.

**Comment by korydraughn** [comment]
If those tests fail regardless of your changes, we should capture those in an issue and investigate them before the next release.

**Comment by d-w-moore** [comment]
Yep think they have more to do with Python RE plugin than iRODS core.

**Comment by alanking** [comment]
https://github.com/irods/irods_rule_engine_plugin_python/issues/130 filed for the most recently failing tests. `test_misc.test_server_respawns_processes__issue_4977` does not exist on 4-2-stable, so we can file that one later if it crops up again. Merging!

**Pull request merged**